src/components/ManageAccountsDrawer/components/ImportAccounts.tsx

```diff
@@ -122,41 +127,13 @@ export const ImportAccounts = ({ chainId, onClose }: ImportAccountsProps) => {
   // But also need to remove accounts that were disabled.
 
   // initial fetch to detect the number of accounts based on the "first empty account" heuristic
-  useEffect(() => {
-    ;(async () => {
-      if (!wallet) return
-
-      let accountNumber = 0
-      const accounts = []
-
-      while (true) {
-        try {
-          const accountResult = await queryClient.fetchQuery(
-            reactQueries.accountManagement.accountIdWithActivity(accountNumber, chainId, wallet),
-          )
-
-          if (!accountResult) break
-
-          const { accountId, hasActivity } = accountResult
-
-          // Add the account before checking if it has activity so user has ability to toggle it if needed.
-          accounts.push({ accountNumber, accountId })
-
-          // If the account has no activity, it's the first empty account.
-          if (!hasActivity) {
-            break
-          }
-        } catch (error) {
-          console.error(error)
-          break
-        }
-
-        accountNumber++
-      }
+  const { data: allAccountIdsWithActivity } = useQuery(
+    accountManagement.allAccountIdsWithActivity(chainId, wallet, NUM_ADDITIONAL_EMPTY_ACCOUNTS),
+  )
 
-      setAccounts(accounts)
-    })()
-  }, [chainId, queryClient, wallet])
+  useEffect(() => {
+    setAccounts(allAccountIdsWithActivity ?? [])
+  }, [allAccountIdsWithActivity])
```

Comment:
Can use a `selector` in the `useQuery()` call to achieve this if the goal is simply to get a default, there's probably no reason to have a state field anymore here

Reply:
Interesting thought, the "load more" functionality needs to mutate the output of the initial paging of accounts, so unless we engineer some sort of `useInfiniteQuery` we'll need to keep this as glue between the two queries. 

Happy to discuss if you think we should go a step further though?